src/proc/namer.rs

```diff
@@ -86,16 +102,12 @@ impl Namer {
         })
     }
 
-    fn namespace(&mut self, reserved_keywords: &[&str], f: impl FnOnce(&mut Self)) {
-        let parent_unique = std::mem::take(&mut self.unique);
-        self.unique.extend(
-            reserved_keywords
-                .iter()
-                .map(|string| (string.to_string(), 0)),
-        );
-
+    fn namespace(&mut self, f: impl FnOnce(&mut Self)) {
+        self.namespace_index += 1;
         f(self);
-        self.unique = parent_unique;
+        let current_ns = self.namespace_index;
+        self.unique.retain(|&(_, ns), _| ns != current_ns);
```

Comment:
how could anything be in here? we just bumped the namespace index, which was never seen before

Reply:
why are we clearing this up at all?

Reply:
This is after the closure call, that is about to add stuff to that hashmap by using `call` method. After all this is the reason tu use `namespace` method in the first place.

The idea is that once you are done with given namespace, you clear it out, so next time the index is bumped, the list is empty.

In old implementation, every time you enter the namespace, a new unique map was created, and it was discarded once you exit the namespace. Here, it's being cleared instead.

Reply:
I don't quite understand why we are able to clear everything. Once you write down a structure with its members, Namer needs to generate their names. But then, anywhere through the backend code, it may need to ask the Namer: "what was the name of this field?" and get that back. If we are clearing the subset, Namer wouldn't be able to answer this.

Reply:
The unique hashmap is only used for collision detection. The result of renames is written down to the `output` map, which is a separate entity inside `reset` method. It has nothing to do with `unique`. Thus, the result of naming is still properly persisted.

Reply:
ah, ok, that's good :)